### ISSUE TITLE
util: parse_time() accepts invalid strings

### ISSUE THREAD

**Issue by yuwata** [systemd/systemd#10619]
E.g.
```
$ systemd-analyze timespan 123.45.67
Original: 123.45.67
      μs: 124120000
   Human: 2min 4.120000s
$ systemd-analyze timespan '123sec 45sec'
Original: 123sec 45sec
      μs: 168000000
   Human: 2min 48s
```
I think `parse_time()` should use a similar logic with `parse_size()`.

**Comment by poettering** [comment]
I think the sectoin example you gave is actually fine to support... Using timespan specifications as a simple calculator for additions is totally fine i think. It's a very systematic extension to accept "4s 5s" if "4s 5000ms" is allowed.

but yeah, the first example should really be fixed, it makes no sense indeed to accept that.

**Comment by yuwata** [comment]
@poettering OK. The first example is fixed by #10648. PTAL.
